api/products.thrift

```diff
@@ -0,0 +1,78 @@
+// -------------------------------------------------------------------------
+//                     The CodeChecker Infrastructure
+//   This file is distributed under the University of Illinois Open Source
+//   License. See LICENSE.TXT for details.
+// -------------------------------------------------------------------------
+
+include "shared.thrift"
+
+namespace py ProductManagement
+namespace js ProductManagement
+
+
+/*
+struct PrivilegeRecord {
+  1: string   name,
+  2: bool     isGroup
+}
+typedef list<PrivilegeRecord> PrivilegeRecords
+*/
+
+struct DatabaseConnection {
+  1:          string engine,         // The database engine, such as "sqlite" or "postgresql".
+  2:          string host,
+  3:          i32    port,
+  4:          string username_b64,
```

Comment:
In the future, no. But right now, Thrift is pretty much messed up and there isn't a good way to get over this. This was a decision to be made. Later on, when the upstream packages are updated, the API will be changed.